core/drivers/stm32_tamp.c

```diff
@@ -203,20 +374,120 @@ static void stm32_tamp_set_secure(struct stm32_tamp_instance *tamp,
 	}
 }
 
-static void stm32_tamp_set_privilege(struct stm32_tamp_instance *tamp,
-				     uint32_t mode)
+static void stm32_tamp_set_privilege(uint32_t mode)
 {
-	vaddr_t base = io_pa_or_va(&tamp->base, 1);
+	vaddr_t base = io_pa_or_va(&stm32_tamp_dev->pdata.base, 1);
 
-	if (tamp->compat && (tamp->compat->tags & TAMP_HAS_REGISTER_PRIVCFGR))
+	if (stm32_tamp_dev->pdata.compat &&
+	    (stm32_tamp_dev->pdata.compat->tags & TAMP_HAS_REGISTER_PRIVCFGR))
 		io_clrsetbits32(base + _TAMP_PRIVCFGR, _TAMP_PRIVCFGR_MASK,
 				mode & _TAMP_PRIVCFGR_MASK);
 }
 
-static TEE_Result stm32_tamp_parse_fdt(struct stm32_tamp_instance *tamp,
-				       const void *fdt, int node,
+static void parse_bkpregs_dt_conf(const void *fdt, int node,
+				  struct stm32_tamp_platdata *pdata)
+{
+	const fdt32_t *cuint = NULL;
+	unsigned int bkpregs_count = 0;
+	int lenp = 0;
+
+	cuint = fdt_getprop(fdt, node, "st,backup-zones", &lenp);
+	if (!cuint) {
+		panic("Missing backup registers configuration");
+		return;
+	}
+
+	pdata->bkpregs_conf = calloc(1, sizeof(*pdata->bkpregs_conf));
+	if (!pdata->bkpregs_conf)
+		panic("No memory available for backup registers RIF config");
+
+	/*
+	 * When TAMP does not support RIF, the backup registers can
+	 * be splited in 3 zones. These zones have specific read/write
+	 * access permissions based on the secure status of the accesser.
+	 * When RIF is supported, these zones can additionally be splited
+	 * in subzones that have CID filtering. Zones/Subzones can be empty and
+	 * are contiguous.
+	 */
+	if (!(stm32_tamp_dev->pdata.compat->tags & TAMP_HAS_RIF_SUPPORT)) {
+		/* 3 zones, 2 offsets to apply */
+		if (lenp != sizeof(uint32_t) * 3)
+			panic("Incorrect bkpregs configuration");
+
+		pdata->bkpregs_conf->nb_zone1_regs = fdt32_to_cpu(cuint[0]);
+		bkpregs_count = fdt32_to_cpu(cuint[0]);
+
+		pdata->bkpregs_conf->nb_zone2_regs = bkpregs_count +
```

Comment:
This changes the meaning of `nb_zone2_regs` which now count the number of registers in zone1 + zone2. This comments applies also to the RIF config case.
I suggest you update the inline description of `struct stm32_bkpregs_conf`.